pkg/kubelet/kubelet.go

```diff
@@ -2953,6 +2954,51 @@ func disallowResizeForSwappableContainers(runtime kubecontainer.Runtime, desired
 	return false, ""
 }
 
+// PodResourceSummary save container and pod-level resources for resizing
+type PodResourceSummary struct {
+	// TODO: Add pod-level resources here once resizing pod-level resources is supported
+	InitContainers map[string]v1.ResourceRequirements `json:"InitContainers,omitempty"`
+	Containers     map[string]v1.ResourceRequirements `json:"Containers,omitempty"`
+}
+
+// getPodResizeInfo to cache containers and pod resources CPU and memory resources once resizing is completed
+func (kl *Kubelet) getPodResizeInfo(allocatedPod *v1.Pod, podStatus *kubecontainer.PodStatus) map[string]string {
+	resizedInfo := &PodResourceSummary{
+		InitContainers: make(map[string]v1.ResourceRequirements),
+		Containers:     make(map[string]v1.ResourceRequirements),
+	}
+
+	podutil.VisitContainers(&allocatedPod.Spec, podutil.InitContainers|podutil.Containers,
+		func(allocatedContainer *v1.Container, containerType podutil.ContainerType) bool {
+			if !isResizableContainer(allocatedContainer, containerType) {
+				// If the container is not resizeable container, it doesn't need to be saved to resize completed event annotations
+				return true
+			}
+
+			containerStatus := podStatus.FindContainerStatusByName(allocatedContainer.Name)
+			if containerStatus == nil || containerStatus.State != kubecontainer.ContainerStateRunning {
+				// If the container isn't running, it doesn't need to be saved to completed event
+				return true
+			}
+
+			actuatedResources, _ := kl.allocationManager.GetActuatedResources(allocatedPod.UID, allocatedContainer.Name)
+			var targetMapContainers map[string]v1.ResourceRequirements
+			if containerType == podutil.InitContainers {
+				targetMapContainers = resizedInfo.InitContainers
+			} else {
+				targetMapContainers = resizedInfo.Containers
+			}
+			targetMapContainers[allocatedContainer.Name] = *actuatedResources.DeepCopy()
```

Comment:
nit: this is immediately being used to marshal to json, and isn't passed outside this function or mutated. I think you can skip the `DeepCopy`